File: kotlin/src/main/kotlin/Binary Search.kt
package o.mg

fun search(nums: IntArray, target: Int): Int {
    var index = -1
    var start = 0
    var end = nums.size - 1
    while (start <= end) {
        val mid = (start + end) / 2
        if (nums[mid] == target) {
            index = mid
            break
        }
        if (nums[mid] < target) {
            start = mid + 1
        } else {
            end = mid - 1
        }
    }
    return index
}

File: go/solutions/binary_search.go
package solutions

func Search(nums []int, target int) int {
	r := -1
	start := 0
	end := len(nums) - 1
	for start <= end {
		mid := (start + end) / 2
		if nums[mid] == target {
			r = mid
			break
		} else if nums[mid] < target {
			start = mid + 1
		} else if nums[mid] > target {
			end = mid - 1
		}
	}
	return r
}
